transient-aws/hive-example/dispatch.sh

```diff
@@ -0,0 +1,316 @@
+    exit -1
+fi
+
+#
+# Cloudera Director server credentials and URL
+#
+
+DIRECTOR_SERVER="http://localhost:7189"
+DIRECTOR_USER=admin
+DIRECTOR_PASSWORD=admin
+
+#REPLACE_ME
+SSH_USERNAME="centos"
+SSH_PRIVATE_KEY="~/.ssh/id_rsa"
+
+#
+# Optional overrides for config entity names
+#
+
+ENVIRONMENT_NAME="AWS-${RANDOM_TOKEN}"
+DEPLOYMENT_NAME="ETL-ON-AWS"
+CLUSTER_NAME="job_${RANDOM_TOKEN}"
+
+GATEWAY_GROUP_NAME="masters"
+PROVISION_CONFIG="None"
+TERMINATE="true"
+KEEP_CM="false"
+
+function usage {
+    cat <<USAGE_TEXT
+Usage: $0 <optional arguments> -f=cluster.conf job-script.sh [file1.jar file2.zip ...]"
+
+Optional arguments:
+
+ -s, --server            server URL (default ${DIRECTOR_SERVER})
+ -u, --user              server API admin user (default ${DIRECTOR_USER})
+ -p, --password          server API admin password
+ -e, --environment       environment name (default ${ENVIRONMENT_NAME})
+ -d, --deployment        deployment name (default ${DEPLOYMENT_NAME})
+ -c, --cluster           cluster name (random by default with prefix job_)
+ -g, --gateway-group     gateway group name (default ${GATEWAY_GROUP_NAME})
+ -n, --ssh-username      SSH username to use to connect to the gateway (default ${SSH_USERNAME})
+ -i, --ssh-private-key   SSH private key to use to  connect to the gateway (default ${SSH_PRIVATE_KEY})
+ -f, --provision-config  Provision a new cluster with the given config file
+ -t, --terminate         Terminate the cluster (default ${TERMINATE})
+ -k, --keep-cm           Keep CM around if --terminate is true (default ${KEEP_CM})
+
+Example usage:
+
+  $0 -u=admin -i=test.pem -f=cluster.conf -t job1.sh data.zip
+
+USAGE_TEXT
+}
+
+for i in "$@"
+do
+case $i in
+    -s=*|--server=*)
+    DIRECTOR_SERVER="${i#*=}"
+    shift
+    ;;
+    -u=*|--user=*)
+    DIRECTOR_USER="${i#*=}"
+    shift # past argument=value
+    ;;
+    -p=*|--password=*)
+    DIRECTOR_PASSWORD="${i#*=}"
+    shift # past argument=value
+    ;;
+    -e=*|--environment=*)
+    ENVIRONMENT_NAME="${i#*=}"
+    shift # past argument=value
+    ;;
+    -d=*|--deployment=*)
+    DEPLOYMENT_NAME="${i#*=}"
+    shift # past argument=value
+    ;;
+    -c=*|--cluster=*)
+    CLUSTER_NAME="${i#*=}"
+    shift # past argument=value
+    ;;
+    -g=*|--gateway-group=*)
+    GATEWAY_GROUP_NAME="${i#*=}"
+    shift # past argument=value
+    ;;
+    -n=*|--ssh-username=*)
+    SSH_USERNAME="${i#*=}"
+    shift # past argument=value
+    ;;
+    -i=*|--ssh-private-key=*)
+    SSH_PRIVATE_KEY="${i#*=}"
+    shift # past argument=value
+    ;;
+    -f=*|--provision-config=*)
+    PROVISION_CONFIG="${i#*=}"
+    shift # past argument=value
+    ;;
+    -t|--terminate)
+    TERMINATE="true"
+    shift
+    ;;
+    -h|--help)
+    usage
+    exit 0
+    shift
+    ;;
+    *)
+          # unknown option
+    ;;
+esac
+done
+
+CLUSTER_URL="${DIRECTOR_SERVER}/api/v4/environments/${ENVIRONMENT_NAME}/deployments/${DEPLOYMENT_NAME}/clusters/${CLUSTER_NAME}"
```

Comment:
It might be overkill, but should the names here be checked to make sure they are OK in URLs?